packages/cli/fern-definition/schema/src/schemas/SingleUnionTypeSchema.ts

```diff
@@ -1,19 +1,23 @@
 import { z } from "zod";
 import { SingleUnionTypeKeySchema } from "./SingleUnionTypeKeySchema";
 import { WithNameAndDocsSchema } from "./WithNameAndDocsSchema";
+import { WithAvailabilitySchema } from "./WithAvailabilitySchema";
 
 export const SingleUnionTypeSchema = z.union([
     z.string(),
-    WithNameAndDocsSchema.extend({
-        type: z.optional(
-            z.union([
-                z.string(),
-                // allow "{}" as syntactic sugar for null
-                z.record(z.never())
-            ])
-        ),
-        key: z.optional(z.union([z.string(), SingleUnionTypeKeySchema]))
-    })
+    WithAvailabilitySchema.extend(
+        WithNameAndDocsSchema.extend({
+            type: z.optional(
+                z.union([
+                    z.string(),
+                    // allow "{}" as syntactic sugar for null
+                    z.record(z.never())
+                ])
+            ),
+            key: z.optional(z.union([z.string(), SingleUnionTypeKeySchema])),
+            "display-name": z.optional(z.string())
```

Comment:
can we add a `WithDisplayNameSchema`, i anticipate we will need to add display names in a lot of places

Reply:
bonus would be to get `HttpEndpointSchema` to leverage `WithDisplayNameSchema` as well https://github.com/fern-api/fern/blob/9f869708557f82f097ab5d5a7255a1d5aa2d256b/packages/cli/fern-definition/schema/src/schemas/HttpEndpointSchema.ts#L17